modules/ROOT/pages/java/java-create-instance.adoc

```diff
@@ -4,31 +4,143 @@ include::_attributes.adoc[]
 endif::[]
 :keywords: Java, create instance
 
+== Calling Java Constructors
 
+Let's suppose we have the following Java class `Person` which belongs to the `com.foo` package:
 
+[source,java,linenums]
+----
+package com.foo;
+
+public class Person {
+
+    private String name;
+    private Integer age;
+
+    public Person() {
+        this.name = "Max Mule";
+        this.age = 10;
+    }
+
+    public Person(String name, Integer age) {
+        this.name = name;
+        this.age = age;
+    }
+}
+----
+
+IMPORTANT: The classes used with the Java Module must be exported (unless they belong to the default package). If not, the
+execution will fail with a JAVA:CLASS_NOT_FOUND error. See xref:4.1@mule-runtime::how-to-export-resources.adoc[How to
+export resources].
 
+We can call these constructors using the `new` operation of the Java Module.
 
-This example shows how to create an instance of a class:
+This example shows how to create an instance of Person using the constructor that does not require parameters:
 
 [source,xml,linenums]
 ----
 <java:new class="com.foo.Person" constructor="Person()"/>
 ----
 
-This example shows how to create an instance using the default constructor. You can also use a non-custom constructor.
+Otherwise, if we want to create a Person object using the constructor that
+requires parameters, we would do:
+
 [source,xml,linenums]
 ----
 <java:new class="com.foo.Person" constructor="Person(String, Integer)">
     <java:args>#[{
-      firstName: 'MG',
-      age: 34
+      name: 'Jhon Doe',
+      age: 30
+    }]</java:args>
+</java:new>
+----
+
+For the constructor parameters, you can also specify the full package name, for example
+`constructor="Person(java.lang.String, java.lang.Integer)"`. Of course, this is not needed, but it can
+be useful it you want more clarity in your code or you have clashing class names in your Java code.
+
+
+[IMPORTANT]
+===============================
+When configuring the constructor arguments in the `args` parameter,
+the keys of the map will determine how the parameters will be passed to the constructor. *In
+order to reference the parameters by name* (`firstName`, `age`, etc.),
+the java class containing the Method or Constructor has to be compiled
+using https://docs.oracle.com/javase/tutorial/reflect/member/methodparameterreflection.html[the `-parameters` compiler flag].
+If the class was not compiled with this flag, then the same parameters
+must be referenced in the declared order and with the canonical names
+(`arg0`, `arg1`, etc.). In this case:
+[source,xml,linenums]
+----
+<java:args>#[{
+  arg0: 'Jhon Doe',
+  arg1: 30
+}]</java:args>
+----
+
+If your Java classes are defined in your Studio Project, you will need to add
+the Maven Compiler plugin as a dependency in you `pom.xml` in order to compile your classes
```

Comment:
`you will need to add the Maven Compiler plugin as a dependency in you pom.xml` -> `is needed to add the Maven Compiler plugin in your pom.xml` 
Is not a dependency, is a plugin configuration